lib/Basic/SignalHandlerWrapper.cpp

```diff
@@ -0,0 +1,57 @@
+//===-- SignalHandlerWrapper.cpp ------------------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llbuild/Basic/SignalHandlerWrapper.h"
+
+#include <signal.h>
+
+using namespace llbuild;
+using namespace llbuild::basic;
+using namespace llbuild::basic::sys;
+
+SignalHandlerWrapper::SignalHandlerWrapper() {
+#if !defined(_WIN32)
+  innerAction = new struct sigaction();
+#endif
+}
+
+SignalHandlerWrapper::SignalHandlerWrapper(Handler handler)
+    : SignalHandlerWrapper() {
+#if defined(_WIN32)
+  innerAction = handler;
+#else
+  struct sigaction action {};
+  action.sa_handler = handler;
+  *static_cast<struct sigaction *>(innerAction) = action;
+#endif
+}
+
+SignalHandlerWrapper::~SignalHandlerWrapper() {
+#if !defined(_WIN32)
+  delete static_cast<struct sigaction *>(innerAction);
+#endif
+}
+
+void SignalHandlerWrapper::registerAction(int sig,
+                                          SignalHandlerWrapper *previous) {
```

Comment:
I think it would make more sense to just take these arguments on the constructor, wouldn't it?